### PR TITLE
Create user from cli

### PR DESCRIPTION
**Pull request by Sunnepah**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/3428/

File: appscale/tools/appscale.py [hunk 1]
```diff
@@ -311,6 +313,27 @@ def up(self):
...
+
+        Raises:
+          AppScalefileException: If there is no AppScalefile in the current
+            directory.
+          TypeError: If the user does not provide an new user credentials
+        """
+    contents = self.read_appscalefile()
+    contents_as_yaml = yaml.safe_load(contents)
+
+    # construct the appscale-gather-logs command
```
Comment (whoarethebritons) [inline appscale/tools/appscale.py]: Leftover from a copy paste?

File: appscale/tools/appscale.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ class AppScale():
   deploy <app>                      Deploys a Google App Engine app to AppScale:
                                     <app> can be the top level directory with the
                                     code or a tar.gz of the source tree.
+  create-user [--admin]             Create a new user. If --admin option is specified,
```
Comment (tmarballi) [inline appscale/tools/appscale.py]: Could you change this to "Creates a new user"?

File: appscale/tools/scripts/appscale.py [hunk 3]
```diff
@@ -77,6 +77,24 @@ def main():
...
+    try:
+      if len(sys.argv) < 2 or len(sys.argv) > 3:
+        cprint("Usage: appscale create-user [--admin]", 'red')
+        sys.exit(1)
+      if len(sys.argv) == 3:
+        if sys.argv[2] == '--admin':
+          appscale.create_user(True)
+        else:
+          cprint("Error: Invalid argument to 'create-user' command. To create user as admin, "
+                 "you should specified the option '--admin'", 'red')
```
Comment (whoarethebritons) [inline appscale/tools/scripts/appscale.py]: `you should specify the option`

File: appscale/tools/appscale.py [hunk 4]
```diff
@@ -311,6 +313,27 @@ def up(self):
 
     AppScaleTools.run_instances(run_instances_opts)
 
+  def create_user(self, is_admin=False):
+    """ 'create_user' create a new user from the CLI
```
Comment (tmarballi) [inline appscale/tools/appscale.py]: Could you change this docstring to "Creates a new user from the tools."

File: appscale/tools/scripts/appscale.py (hunk 3 above)
Comment (tmarballi) [inline appscale/tools/scripts/appscale.py]: Please change "specified" to "specify".

**Comment by obino** [comment]
@Sunnepah would you mind to pull in the latest master? We will need the new 3.3 tag with your code to ensure ti goes through the regression testing.

**Comment by obino** [comment]
@Sunnepah would you mind to pull in the latest master? We will need the new 3.3 tag with your code to ensure ti goes through the regression testing.

File: appscale/tools/appscale.py (hunk 1 above)
Comment (Sunnepah) [inline appscale/tools/appscale.py]: Exactly, good catch. :)

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3460/

File: appscale/tools/appscale_tools.py [hunk 5]
```diff
@@ -635,6 +635,36 @@ def reset_password(cls, options):
...
+        """
+    secret = LocalState.get_secret_key(options.keyname)
+    login_host = LocalState.get_login_host(options.keyname)
+
+    username, password = LocalState.get_credentials(is_admin)
+    encrypted_password = LocalState.encrypt_password(username, password)
+
+    acc = AppControllerClient(login_host, secret)
+
+    RemoteHelper.create_user_accounts(username, encrypted_password, login_host, options.keyname)
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: `create_user_accounts` already hashes the password, so you should just provide the plaintext password as the argument to this function.

**Comment by cdonati** [comment]
Thanks, @Sunnepah!

**Comment by Sunnepah** [comment]
👍

**Pull request merged**